### PR TITLE
fix(scanner): stop flagging homepages as edited on every scan

### PR DESCRIPTION
**Pull request by tessak22**
## Problem
Every competitor's homepage (and any other page with a previous scan) is being flagged as "edited" on every scan. The new `/[slug]/history` UI exposed the issue but the root cause is older.

## Root cause — `generate.json` double-fetches
`lib/scanner.ts` calls `generateDiff()` with `url: input.url` and `previousContent: toPreviousContent(previousScan)`. Tabstack's `generate.json` **re-fetches** that URL server-side — so the diff LLM compares:

- **Previous**: the stored markdown from the last scan
- **"Current"**: a *fresh* fetch Tabstack just did — seconds after the primary scan, possibly a different render

Any dynamic element (rotating hero copy, timestamps, A/B variants, session tokens) produces a "changed" verdict on every run. This has likely been happening since the scanner was introduced; the new history page just makes it visible.

## Fix
Two complementary changes:

1. **Short-circuit on matching content** (`lib/scanner.ts`)
   After `runPrimaryScan`, normalize the previous and current markdowns (CRLF → LF, trim trailing whitespace, collapse extra blank lines) and compare as strings. When equal → skip the `generate.json` diff call entirely and persist the scan with `hasChanges=false`, `diffSummary=null`. Saves an LLM round-trip on every no-op scheduled scan.

2. **`currentContent` passthrough** (`lib/tabstack/generate.ts`)
   When content genuinely differs, pass the just-captured markdown as a new `currentContent` field on `GenerateDiffInput`. The prompt is rewritten to instruct the model to compare **only** the two provided snapshots, ignoring any live fetch that `generate.json` performs alongside the request. The original single-version prompt is preserved as a fallback when `currentContent` isn't supplied (backward compatible with any other caller).

## Tests
- New: short-circuits when previous and current markdown match (even with differing line endings / trailing whitespace) — asserts `generateDiff` is not called and the scan row stores `hasChanges=false, diffSummary=null`.
- New: when content genuinely differs, `generateDiff` is called with both `previousContent` and `currentContent`.
- All 357 existing tests pass; `npm run typecheck` clean.

## Test plan (post-merge)
- [ ] Trigger a manual scan of an unchanged competitor page; confirm `hasChanges=false` in DB and the history row shows "no change."
- [ ] Trigger a scan after making a real change; confirm diff summary still renders correctly.
- [ ] Check API logs — `generate` calls should drop noticeably after this ships (no-op scans no longer call it).

## Related
Separately tracked DX note in `notes-local/tabstack-dx-notes.md` (gitignored) on the `generate.json` re-fetch behavior for upstream SDK feedback.

### CODE REVIEW

File: lib/tabstack/generate.ts [hunk 1]
```diff
@@ -100,7 +109,33 @@ export async function generateDiff(input: GenerateDiffInput): Promise<GenerateJs
     );
   }
 
-  const instructions = `Compare these two versions of a competitor page.
+  const rawCurrentContent = input.currentContent;
+  const hasCurrentContent = typeof rawCurrentContent === "string" && rawCurrentContent.length > 0;
```
Comment (tessak22) [inline lib/tabstack/generate.ts]: Blocking: `currentContent` is treated as present only when `length > 0`. If the new snapshot is intentionally empty (`""`), this falls back to the live-fetch prompt path, which can reintroduce false positives and also misses the valid "everything removed" diff case. Treat any string (including empty) as an authoritative snapshot and add a regression test for the empty-string path.

**Review by tessak22 — COMMENTED** [review]
Strict review of latest head (`f996607`).

Result: 1 blocking finding.

Blocking:
- `lib/tabstack/generate.ts`: `currentContent` is considered present only when `length > 0`. For an intentionally empty snapshot (`""`), the code falls back to live-fetch comparison behavior, which can reintroduce false positives and misses the valid full-removal diff case.

Requested fix direction:
- Treat any provided string (including empty string) as authoritative `currentContent`, and add a regression test covering the empty-string path.

Notes:
- The normalized short-circuit in `scanner.ts` is a solid improvement for whitespace-only churn.

File: lib/tabstack/generate.ts (hunk 1 above)
Comment (tessak22) [reply lib/tabstack/generate.ts]: Addressed in 81bc54e — guard changed to `typeof rawCurrentContent === "string"` (dropped the length check). Empty-string currentContent now routes to the two-version prompt, preserving the "everything removed" diff case and keeping the live-fetch fallback off. Also switched the instructions ternary from testing `currentContent` (falsy on `""`) to `hasCurrentContent`. Two regression tests added: empty-string currentContent still picks the two-version prompt and preserves the previous marker; full previous + current pair injects both markers.

**Review by tessak22 — COMMENTED** [review]
Re-review of latest head (`81bc54e`).

Status: previous blocking issue is resolved.

What I re-checked:
- `generateDiff` now treats empty-string `currentContent` as authoritative
- Added regression tests for empty-string and explicit two-snapshot prompt path
- Existing scanner and generate targeted tests pass locally

Validation:
- `npx vitest run lib/__tests__/scanner.test.ts lib/tabstack/__tests__/generate.test.ts` passed (33 tests).

Result: no findings in this pass.
